bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastAudioSink.java

```diff
@@ -30,29 +36,43 @@
  * @author Jason Holmes - Initial contribution
  */
 @NonNullByDefault
-public class ChromecastAudioSink {
+public class ChromecastAudioSink extends AudioSinkAsync {
     private final Logger logger = LoggerFactory.getLogger(ChromecastAudioSink.class);
 
+    private static final Set<AudioFormat> SUPPORTED_FORMATS = Set.of(AudioFormat.MP3, AudioFormat.WAV);
+    private static final Set<Class<? extends AudioStream>> SUPPORTED_STREAMS = Set.of(AudioStream.class);
+
     private static final String MIME_TYPE_AUDIO_WAV = "audio/wav";
     private static final String MIME_TYPE_AUDIO_MPEG = "audio/mpeg";
 
-    private final ChromecastCommander commander;
+    private final ChromecastHandler handler;
     private final AudioHTTPServer audioHTTPServer;
     private final @Nullable String callbackUrl;
 
-    public ChromecastAudioSink(ChromecastCommander commander, AudioHTTPServer audioHTTPServer,
+    public ChromecastAudioSink(ChromecastHandler handler, AudioHTTPServer audioHTTPServer,
             @Nullable String callbackUrl) {
-        this.commander = commander;
+        this.handler = handler;
         this.audioHTTPServer = audioHTTPServer;
         this.callbackUrl = callbackUrl;
     }
 
-    public void process(@Nullable AudioStream audioStream) throws UnsupportedAudioFormatException {
+    @Override
+    public String getId() {
+        return handler.getThing().getUID().toString();
+    }
+
+    @Override
+    public @Nullable String getLabel(@Nullable Locale locale) {
+        return handler.getThing().getLabel();
+    }
+
+    @Override
+    public void processAsynchronously(@Nullable AudioStream audioStream) throws UnsupportedAudioFormatException {
         if (audioStream == null) {
             // in case the audioStream is null, this should be interpreted as a request to end any currently playing
             // stream.
             logger.trace("Stop currently playing stream.");
-            commander.handleCloseApp(OnOffType.ON);
+            handler.stop();
```

Comment:
You should try to close the audio stream in this case.

Reply:
Not here, as audioStream is null.

Reply:
I am sorry, I put my comment on the wrong line. It is expected at line 82 when you handle URLAudioStream.

Reply:
I should have seen it. Thanks. Done.